pkg/blobcache/blobcache.go

```diff
@@ -219,7 +221,7 @@ func (s *blobCacheSource) GetManifest(ctx context.Context, instanceDigest *diges
 }
 
 func (s *blobCacheSource) HasThreadSafeGetBlob() bool {
-	return false
+	return true
```

Comment:
This needs to respect the `HasThreadSafeGetBlob` value of the underlying source.